islandora_scholar.drush.inc

```diff
@@ -118,3 +134,147 @@ EOQ;
+    drush_backend_batch_process();
+  }
+  else {
+    drush_user_abort(dt('Exiting, no datastreams changed.'));
+    exit;
+  }
+}
+
+/**
+ * Batch info for drush command to update citations with HTML in MODS.
+ */
+function islandora_scholar_fix_doi_html_in_mods_create_batch($action) {
+  return array(
+    'operations' => array(
+      array('islandora_scholar_fix_doi_html_in_mods_batch_operation', array($action)),
+    ),
+    'title' => t('Updating MODS for citations...'),
+    'init_message' => t('Preparing to update MODS.'),
+    'progress_message' => t('Time elapsed: @elapsed <br/>Estimated time remaining @estimate.'),
+    'error_message' => t('An error has occurred.'),
+    'file' => drupal_get_path('module', 'islandora_scholar') . '/islandora_scholar.drush.inc',
+  );
+}
+
+/**
+ * Batch operations for drush command to update citations with HTML in MODS.
+ */
+function islandora_scholar_fix_doi_html_in_mods_batch_operation($action, &$context) {
+  $batch_size = 10;
+  /* The islandora_doi_crossref_translator, which introduced the bug that this
+  fixes, is part of the DOI importer, which is hard-coded to
+  import only ir:citationCModel objects. If the repository manager has
+  changed the content models of imported objects, they are responsible for
+  finding and fixing any objects that need it.
+  */
+  $query = <<<EOQ
+SELECT ?pid FROM <#ri>
+WHERE {
+  ?pid <fedora-model:hasModel> <info:fedora/ir:citationCModel> ;
+       <fedora-view:disseminates> ?ds .
+  ?ds <fedora-view:disseminationType> <info:fedora/*/MODS> .
+}
+ORDER BY ?pid
+EOQ;
+  $connection = islandora_get_tuque_connection();
+
+  $sandbox = &$context['sandbox'];
+  if (!isset($sandbox['offset'])) {
+    $sparql_count = $connection->repository->ri->countQuery($query, 'sparql');
+    $sandbox['offset'] = 0;
+    $sandbox['total'] = $sparql_count;
+    if ($sandbox['total'] === 0) {
+      return;
+    }
+  }
+
+  $context['message'] = t('Processing results @start to @end.', array(
+    '@start' => $sandbox['offset'],
+    '@end' => min($sandbox['offset'] + $batch_size, $sandbox['total']),
+  ));
+
+  $offset_start = $sandbox['offset'];
+  $query .= "
+  LIMIT $batch_size
+  OFFSET $offset_start
+  ";
+  $results = $connection->repository->ri->sparqlQuery($query);
+  module_load_include('inc', 'islandora_doi', 'includes/utilities');
+  foreach ($results as $result) {
+    $object = islandora_object_load($result['pid']['value']);
+    $namespace = 'http://www.loc.gov/mods/v3';
+    $doc = new DOMDocument();
+    $doc->loadXML($object['MODS']->content);
+    $needs_work = FALSE;
+    /* The crossref metadata spec dictates that 'Face markup' can only be
+    present in title, subtitle, original_language_title, and
+    unstructured_citation (not implemented in the doi translator).
+    src: https://support.crossref.org/hc/en-us/articles/214532023
+    */
+    foreach (array('title', 'subtitle') as $local_name) {
```

Comment:
Since we mend `original_language_title` in [`modules/doi/includes/utilities.inc`](https://github.com/Islandora/islandora_scholar/pull/288/files#diff-52bb712f031b7c5151fc17f7508f7213R337), we probably should do it also here, no? If yes, [that line](https://github.com/Islandora/islandora_scholar/pull/288/files#diff-d342542a4ca9df72fe6804cad9aa55dbR239) should read
```php
    foreach (array('title', 'subtitle', 'original_language_title') as $local_name) {
```

Reply:
@d-r-p @rosiel I'm also not sure if `subtitle` is a typo and should actually be `subTitle` (capital T for the correct MODS element name), or if this is doing something else I don't understand.

Reply:
local names are case sensitive, so yes, your concern @bryjbrown is valid. if MODS defines subTitle we need to check for subTitle. @bryjbrown since that change is already merged in your branch, could you fix it? Really no need to wait for @rosiel, that is totally normal workflow.

Reply:
Yeah sorry about that, forgot to comment that I goofed and yes it's subTitle in MODS; i didn't check carefully enough (it's subtitle in crossref). 

But since this is in the Drush that's editing MODS, original_language_title isn't a thing in MODS. It would be titleInfo[@type="translated"]/title that we'd look for.